documentation/website/src/handbook/integration/serenity-2-migration-guide.md

```diff
@@ -0,0 +1,688 @@
+
+```javascript
+import { AnswersQuestions, Question, UsesAbilities } from "@serenity-js/core";
+import { WalksTargets } from "../../ui/ramblers/walksTargets";
+
+export class UploadError implements Question<Promise<boolean>> {
+
+  static displayed = () => new UploadError();
+
+  toString() {
+    return "walk upload error is showing";
+  }
+
+  answeredBy(actor: AnswersQuestions & UsesAbilities): Promise<boolean> {
+    return WalksTargets.uploadResultSummary.answeredBy(actor).isPresent();
+  }
+
+}
+```
+
+So to get over this, you will need to wrap the returned statement in `promiseOf`. This function resolves the `wdpromise.Promise`, then returns that value in a new native Promise. 
+So, to get the code to compile it will now look like this: 
+
+```javascript
+import { AnswersQuestions, Question, UsesAbilities } from "@serenity-js/core";
+import { promiseOf } from "@serenity-js/protractor/lib/promiseOf";
+import { WalksTargets } from "../../ui/ramblers/walksTargets";
+
+export class UploadError implements Question<Promise<boolean>> {
+
+  static displayed = () => new UploadError();
+
+  toString() {
+    return "walk upload error is showing";
+  }
+
+  answeredBy(actor: AnswersQuestions & UsesAbilities): Promise<boolean> {
+    return promiseOf(WalksTargets.uploadResultSummary.answeredBy(actor).isPresent());
+  }
+
+}
+```
+
+If you are still having compilation problems after doing the above, then you will need to provide the Typescript compiler some help that this method is fine by adding `as Promise<boolean>` (or whatever the return type is from your `Question`) to the end of the return statement.
+If your method returns a more complex domain type e.g. `Promise<TradeSummary[]>` then adding `as Promise<TradeSummary[]>` may still not work. If this is the case 
+then try adding `as Promise<any>` instead to see whether your code now compiles. This is obviously not ideal as you are essentially removing type-safety from your code, but you can at least see whether the method now compiles, or another problem area is highlighted. 
+ 
+## Migrating Activity code
+### Migrating @step annotations
+
+If you had step annotations in any of your `Activity` classes, like this:
+
+```javascript
+@step("{0} deletes all records")
+```
+
+...then you will have to re-implement the with an equivalent `toString()` method as `@step` has been deprecated in version 2.x. 
+An example of a `toString()` method is as follows :  
+
+```javascript
+toString() {
+    return "#actor deletes all records";
+  }
+```
+
+There is no quick way of doing this as it requires the original string passed to the annotation to be cut from its original location and appended to the return statement. 
+Be sure to replace the actor placeholder `{0}` with `#actor` in the final string. You will also have to remove the step import as well:  
+  
+```javascript
+import { step } from "@serenity-js/core/lib/recording";
+```
+
```

Comment:
It might be good to mention the [`Task.where`](https://serenity-js.org/modules/core/class/src/screenplay/Task.ts~Task.html#static-method-where) here too.